prototype2/pubsub/src/lib.rs

```diff
@@ -0,0 +1,325 @@
+extern crate specs;
+extern crate uuid;
+extern crate typemap;
+extern crate itertools;
+
+#[macro_use]
+extern crate mopa;
+
+use mopa::Any;
+use std::any::TypeId;
+use std::collections::HashMap;
+use std::marker::PhantomData;
+use std::sync::RwLockWriteGuard;
+
+/**
+ * Knows the specs resource identity of a PubSubToken resource
+ */
+pub struct PubSubSystem {
+  type_to_subscriber: HashMap<TypeId, Subscriber>,
```

Comment:
This should be keyed on (TypeId, SubscriberIdx). That will let us have broadcast